plugins/scanners/scancode/src/main/kotlin/ScanCodeResultParser.kt

```diff
@@ -61,11 +61,17 @@ private fun parseResult(result: JsonElement): ScanCodeResult {
             polymorphicDefaultDeserializer(CopyrightEntry::class) { CopyrightEntry.Version2.serializer() }
         }
 
-        else -> SerializersModule {
+        3 -> SerializersModule {
             polymorphicDefaultDeserializer(FileEntry::class) { FileEntry.Version3.serializer() }
             polymorphicDefaultDeserializer(LicenseEntry::class) { LicenseEntry.Version3.serializer() }
             polymorphicDefaultDeserializer(CopyrightEntry::class) { CopyrightEntry.Version2.serializer() }
         }
+
+        else -> SerializersModule {
```

Comment:
Should the else be turned into `4 -> ` for clarifying that this is support for version 4 but not beyond?

Reply:
Two reasons why I maintained the `else`:
1. There is a chance that new major versions still work if the major changes are introduced in parts we do not care about, like in the `dependencies` structure. That's also why we only warn if the major version is exceeded.
2. We need an `else` anyway to make the `when` exhaustive.

Reply:
> There is a chance

For a component where correctness matters as much as for the license scan it might be reasonable to not rely on a "maybe it works".